api/admin/controller/__init__.py

```diff
@@ -415,20 +417,29 @@ def _sort_by_collection(self, timestamps):
         collection ID and each value is a list of the timestamps associated with that collection."""
 
         result = {}
+        collection_name = None
         for timestamp in timestamps:
-            result.setdefault((timestamp.get("collection_id") or -1), []).append(timestamp)
+            result.setdefault(timestamp.get("collection_name"), []).append(timestamp)
```

Comment:
This is a good solution. I had to check that Collection.name is unique, but it is, so this is good.